lib/Adafruit_HMC5883_Unified/Adafruit_HMC5883_U.cpp

```diff
@@ -111,26 +111,30 @@ void Adafruit_HMC5883_Unified::read() {
   Wire.endTransmission();
   Wire.requestFrom((byte)HMC5883_ADDRESS_MAG, (byte)6);
 
-  // Wait around until enough data is available
-  while (Wire.available() < 6)
```

Comment:
This is really bad code, which may deserve a PR at the original maintainer.

Reply:
It is already changed there, but the ARDUINO >= 100 check is still there, no real need to get their code here, AFAICS.